File: native/ConceprFirstRxSwift/rxProxy/DisposableImpl.swift

import Foundation
import RxSwift
import SharedCode

class DisposableImpl : SharedCode.Disposable {
    
    init(disposable: RxSwift.Disposable) {
        self.disposable = disposable
    }
    
    private var disposable: RxSwift.Disposable
    
    func dispose() {
        disposable.dispose()
    }

    func isDisposed() -> Bool {
        return false
    }
}

File: app/src/main/java/rxProxy/DisposableImpl.kt
package rxProxy

import com.jetbrains.handson.mpp.mobile.the_rx.Disposable

class DisposableImpl(
    private val disposable: io.reactivex.disposables.Disposable
) : Disposable {

    override fun dispose() {
        disposable.dispose()
    }

    override fun isDisposed(): Boolean {
        return disposable.isDisposed
    }
}